tiles/src/main/java/com/protomaps/basemap/layers/Roads.java

```diff
@@ -35,98 +43,134 @@ public void processFeature(SourceFeature sourceFeature, FeatureCollector feature
-        highway.equals("road") || highway.equals("raceway")) {
-        feat.setAttr("pmap:kind", "minor_road").setZoomRange(12, 15)
-          .setAttrWithMinzoom("ref", shield_text, 12)
-          .setAttrWithMinzoom("ref_length", shield_text_length, 12)
-          .setAttrWithMinzoom("network", network_val, 12);
-        OsmNames.setOsmNames(feat, sourceFeature, 14);
+        highway.equals("road") || highway.equals("raceway")
+      ) {
+        kind = "minor_road";
+        min_zoom = 12;
+        min_zoom_shield_text = 12;
+        min_zoom_names = 14;
+
+        if (highway.equals("service")) {
+          kind_detail = "service";
+          min_zoom = 13;
+
+          // push down "alley", "driveway", "parking_aisle", "drive-through" & etc
+          if( sourceFeature.hasTag("service") ) {
+            min_zoom = 14;
+            service = sourceFeature.getString("service");
+          }
+        }
       } else if (sourceFeature.hasTag("highway", "pedestrian", "track", "path", "cycleway", "bridleway", "footway",
-        "steps", "corridor")) {
-        feat.setAttr("pmap:kind_detail", highway).setZoomRange(12, 15)
-          .setAttrWithMinzoom("ref", shield_text, 12)
-          .setAttrWithMinzoom("ref_length", shield_text_length, 12)
-          .setAttrWithMinzoom("network", network_val, 12)
-          .setAttr("pmap:kind", "path").setZoomRange(12, 15);
-        OsmNames.setOsmNames(feat, sourceFeature, 14);
+        "steps", "corridor")
+      ) {
+        kind = "path";
+        kind_detail = highway;
+        min_zoom = 12;
+        min_zoom_shield_text = 12;
+        min_zoom_names = 14;
+
+        if( sourceFeature.hasTag("highway", "path", "cycleway", "bridleway", "footway", "steps") ) {
+          min_zoom = 13;
+        }
+        if( sourceFeature.hasTag("footway", "sidewalk", "crossing") ) {
+          min_zoom = 14;
+          kind_detail = sourceFeature.getString("footway");
+        }
+        if( sourceFeature.hasTag("highway", "corridor") ) {
+          min_zoom = 14;
+        }
       } else {
-        feat.setAttr("pmap:kind", "other").setZoomRange(14, 15)
-          .setAttrWithMinzoom("ref", shield_text, 14)
-          .setAttrWithMinzoom("ref_length", shield_text_length, 14)
-          .setAttrWithMinzoom("network", network_val, 14);
-        OsmNames.setOsmNames(feat, sourceFeature, 14);
+        kind = "other";
+        kind_detail = sourceFeature.getString("service");
+        min_zoom = 14;
+        min_zoom_shield_text = 14;
+        min_zoom_names = 14;
+      }
+
+      var feat = features.line("roads")
+                  .setId(FeatureId.create(sourceFeature))
+                  // Core Tilezen schema properties
+                  .setAttr("pmap:kind", kind)
+                  .setAttr("pmap:kind_detail", kind_detail)
+                  .setAttrWithMinzoom("ref", shield_text, min_zoom_shield_text)
+                  .setAttrWithMinzoom("shield_text_length", shield_text_length, min_zoom_shield_text)
```

Comment:
Switch to Tilezen forward naming on the new v3 property for shield_text_length (instead of ref_length).

_NOTE: This isn't a breaking change because we haven't tagged v3 yet (even though `ref_length` was merged for a few weeks._